core/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -346,4 +348,12 @@ package object internal {
       }
     }
   }
+
+  private[http4s] def javaMajorVersion[F[_]](implicit F: Sync[F]): F[Option[Int]] =
+    F.delay(sys.props.get("java.version")).map {
+      case Some(v) =>
+        Try(v.split("\\.").head.toInt).toOption
+      case None =>
+        None
+    }
```

Comment:
I think on Java 8 `java.version` is formatted like `1.8...`.

Reply:
Yes, that's true.  It behaves correctly for the versions I cared about, but it's not yet right in general.